README.md
Material Design UI for Corona Labs SDK

This README is just an overview document. You can find more detailed documentation within the repo in future updates.

What is material-ui?
--------------

A loosely based Material UI library for Corona Labs SDK. It is written in Lua using the free edition of the SDK. The library will help build a UI based on Material Design.

* Requires Corona SDK build 2017.3135 or greater.

* Supports: iOS 9+, Android 6.x+ and other Corona SDK supported platforms.

Installing material-ui from the repo on GitHub
--------------

* Clone the repo or download archive
```
git clone git://github.com/arcadefx/material-ui.git
```
* Copy the required folders into your project:
```
materialui (folder)
icon-font (folder)
```
* Read config.lua to copy to your config.lua the "MUI SET UP" section and also follow the 'content' section for specifying width and height.

Using material-ui
--------------

Warning: The latest update could break your code if you used an older copy. I have not updated the documentation yet. I will get it updated :).

* If using the GitHub repo, edit your scene file and be sure to include at the top:
```
local mui = require( "materialui.mui" )
local muiData = require( "materialui.mui-data" )
```

* In the scene create function add in the initializer and any user-interface elements
```
    mui.init()
    mui.newRoundedRectButton({
        name = "newGame",
        text = "New Game",
        width = 150,
        height = 30,
        x = 100,
        y = 100,
        font = native.systemFont,
        fillColor = { 0, 0.82, 1 },
        textColor = { 1, 1, 1 },
        callBack = mui.actionForButton
    })
```
* In the scene destroy function add in the destroy method to remove all user-interface elements in the scene.
```
    mui.destroy()
```

Building for Device
-------------
Due to device keyboard possibly covering up input fields, be sure to include "coronaWindowMovesWhenKeyboardAppears=true" into iOS settings->iphone->plist table and Android settings.

For an example, see [build.settings](https://github.com/arcadefx/material-ui/blob/master/build.settings) file.

- For iOS devices, please add to you "plist": UILaunchStoryboardName = "LaunchScreen",
- For iOS devices, copy the folder "LaunchScreen.storyboardc" to your project ONLY if it doesn't

Try a Demo
-------------
Using Corona Simulator open up the "main.lua" file in the folder.

Sample Screenshot
-------------
![Alt text](http://www.anedix.com/images/github/material-ui-main.png "Controls including text input")
- Note: The text input is the "Hello, world!"
- Video: https://youtu.be/c8p3DMA6PzU

Available Methods and Documentation
-------------
* See Documentation. [View Documentation](https://corona-material-ui.sourceforge.io/) for more information on methods, examples and properties.

Contributing
-------------
* Feel free to contribute code, testing and feedback.
* Once we get additional authors they will be included in the repo and get recognition for their efforts.
* Please follow the licensing terms for any software included.
* See materialui/mui-example.lua for creating additional modules or review any mui-[name].lua module to see the format.

Contributors (by code, testing, helping, etc):
-------------
    willcastillo
    bluetardis
    StevenWarren
    bodily11
    taigak
    lmy46


Change Log
-------------
* Please see "CHANGELOG.md" in this repo for information on latest changes.

To-Do
-------------
* Expand the library (new widgets: card, clock, calendar, etc)
* Develop a layout engine 
* Develop a palette engine

Summary
-------------
There are also many other files not described here,  please review the .lua files for additional information. :-)

Enjoy!


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file.

## [0.5.0] - 2019-10-16
### Changes
- SVG support and other additions. Will update this file soon.

## [0.3.4] - 2018-01-18
### Changes
- Added SVG support to newCheckBox(), newCircleButton(), newIconButton(), newRadioButton(), newRadioGroup(), newRectButton() and newRoundedRectButton(). Includes 'off' and 'on' states for checkboxes and radio buttons. Demo coming soon with the changes. Documentation will need updated.
- Improved implementation of SVG support.
- To-do: Include 'off' and 'on' states for buttons in general. If you press and hold the button stays in the 'on' state until released. If checkbox or radio button the state remains until touched again.

## [0.3.3] - 2018-01-11
### Changes
- Added SVG graphic support module. It's being integrated into the widgets. This will allow the developer
to support SVG images for Icons. Material font codepoints will remain as-is and SVG will be an option per widget. Currently only newIconButton() uses it.
- Fixed newTextBox() issue when less text is put into the box when prior text existed. It would push the text down on the screen.
- Turn on Svg support, activate the 'nanosvg' plugin from CoronaLabs Marketplace. Supply the flag 'useSvg' set to true in the initialization call. Example: mui.init(nil, {parent=self.view, useSvg = true})

## [0.3.2] - 2017-11-13
### Changes
- Added 'ignoreTap' to all buttons to 'ignore' tap event.
- Improved support for inset settings for display (iPhone X)
- Detecting if Phone or Tablet using mui.isPhone() returns boolean true/false.
- Fixed Cards

## [0.3.1] - 2017-10-13
### Changes
- Added insertRowsTableView() appends rows to an existing table view. Arguments are name of tableView and a list of entries. See documentation for newTableView with Helpers.
- Added removeAllRowsFromTableView() remove all rows from an existing table view.  Argument is the name of the tableView. See documentation for newTableView with Helpers.
- Created a Pagination demo of sqlite database entries using the above methods. See http://www.anedix.com/fileshare/table-paginator.zip and look at file table-paginator.lua.  Note it includes a small database abstraction layer I created. Feel free to use it for other things and it will be improved with caching, etc.
- Updated demo for showing how to attach things to a tableView too. See http://www.anedix.com/fileshare/checklist.zip

## [0.3.0] - 2017-10-05
### Changes
- WARNING: getScaleVal() has been depreciated. Scaling is now fixed and works with config.lua.  See config.lua for more information in demo.  Any existing code based on it needs updated to adjust sizes and placement (x, y). Sorry, it was for the better.
- With scaling fixed, the material design font letters show up and such on various devices now. There was an issue where these did not show correctly.
- Requires Corona SDK build 2017.3135.
- Supports safe zone insets and iPhone X display area.
- Safe area/zone implemented.
- mui-data must be included at the top of your scene file as it needs it for safeArea values. See menu.lua for example.

	-- mui, place below mui require
	local muiData = require( "materialui.mui-data" )
- For iOS devices, please add to you "plist": UILaunchStoryboardName = "LaunchScreen",
- For iOS devices, copy the folder "LaunchScreen.storyboardc" to your project ONLY if it doesn't already exist. If it exist skip this step!
- To place a background in the safe zone be sure to look at the top of the menu.lua file for an example.
- See menu.lua and fun.lua for examples using the new methods.
- Fixed newSelect() to render correctly.
- Fixed some demo code.
- All instances of getScaleVal() have been removed.

## [0.1.99] - 2017-08-02
### Changes
- newTableView() - Added parameter "valign" to attachToRow(). Allows you to vertically align the object.
- newTableView() - Added parameter "valign" to 'list' and 'columns' which allows you to vertically align the text.
- newTableView() - Documentation updated for the "valign" addition.
- newTableView() - Added parameter "align" to documentation.

## [0.1.98] - 2017-07-21
### Changes
- newTableView() - Added internal method attachBackgroundToRow(..). if the newTableView() options in attribute list now supports 'backgroundImage' which you specify the image to use for the row's background.  Each row has to be set and if not it will use the normal color scheme.

### Fixes
- Adding controls to left / right align on a row didn't work correctly.
- Added return "true" to avoid propagation to other controls for buttons. Added a special case for "newTableView()" which avoids the row's callback being used if a button event occurs on top.

## [0.1.97] - 2017-06-12
### Fixes
- newSelect() - Fixed issue where value was being used in the 'selected' text instead of the text for the button.  Now after choosing a drop down item the 'text' for the item is shown.

## [0.1.96] - 2017-05-05
### Fixes
- newTableView() - Tapping repeatly did not register an event. This is fixed now. Which was an issue on all devices and simulator.

## [0.1.93] - 2017-04-23
### Fixes
- newSlidePanel() - Slide panel would not always slide out fully. This is fixed now.

## [0.1.92] - 2017-04-13
### Changes
- newSelect() displays correctly now using a scrollView. It resides within the scrollView. The drop down shows below or just above the select input area depending on visible scroll area.
- newSlidePanel() demo updated on menu.lua.  It demos a custom callback to avoid the default animation when scene switching. It demos using the animated switching.

## [0.1.90] - 2017-04-12
### Changes

fun.lua
-----------------------------------------------------------------------------------------
--
-- fun.lua
--
-----------------------------------------------------------------------------------------

-- corona
local composer = require( "composer" )
local widget = require( "widget" )

local mui = require( "materialui.mui" )

local scene = composer.newScene()

-- mui
local muiData = require( "materialui.mui-data" )

--------------------------------------------

-- forward declarations and other locals
local screenW, screenH, halfW = display.contentWidth, display.contentHeight, display.contentWidth*0.5

local background = nil
local scrollView = nil

local function destroyDemoText( demoText )
    mui.debug("destroyDemoText called")
    if demoText ~= nil then
        demoText:removeSelf()
        demoText = nil
    end
end

function scene:create( event )

	-- Called when the scene's view does not exist.
	-- 
	-- INSERT code here to initialize the scene
	-- e.g. add display objects to 'sceneGroup', add touch listeners, etc.

	local sceneGroup = self.view

    mui.init()

    -- Gather insets (function returns these in the order of top, left, bottom, right)
    local topInset, leftInset, bottomInset, rightInset = mui.getSafeAreaInsets()

    -- Create a vector rectangle sized exactly to the "safe area"
    local background = display.newRect(
        display.screenOriginX + leftInset, 
        display.screenOriginY + topInset, 
        display.contentWidth - ( leftInset + rightInset ), 
        display.contentHeight - ( topInset + bottomInset )
    )
    background:setFillColor( 1 )
    background:translate( background.width*0.5, background.height*0.5 )
    sceneGroup:insert( background )

    mui.newRoundedRectButton({
        name = "goBack",
        text = "Go Back",
        width = 100,
        height = 30,
        x = 80,
        y = 70,
        font = native.systemFont,
        fillColor = { 0.31, 0.65, 0.03, 1 },
        textColor = { 1, 1, 1 },
        iconText = "arrow_back",
        iconFont = mui.materialFont,
        iconFontColor = { 1, 1, 1, 1 },
        -- iconImage = "1484026171_02.png",
        callBack = mui.actionSwitchScene,
        callBackData = {
            sceneDestination = "menu",
            sceneTransitionColor = { 0, 0.73, 1 },
            sceneTransitionAnimation = true
        } -- scene menu.lua
    })

    -- show a "toast" message, yes I said toast like HTML 5 Toast
    -- recommend 40 percent ratio for one liners (20 radius/50 height = 0.40)
    local showToast = function()
        mui.newToast({
            name = "toast_demo",
            text = "New Messages!",
            radius = 20,
            width = 150,
            height = 30,
            font = native.systemFont,
            fontSize = 18,
            fillColor = { 0, 0, 0, 1 },
            textColor = { 1, 1, 1, 1 },
            top = 40 + muiData.safeAreaInsets.topInset,
            easingIn = 500,
            easingOut = 500,
            callBack = mui.actionForButton
        })
    end

    mui.newRoundedRectButton({
        name = "newToast",
        text = "Show Toast",
        width = 100,
        height = 30,
        x = 80,
        y = 110,
        font = native.systemFont,
        textColor = { 1, 1, 1, 1 },
        fillColor = { 0.63, 0.81, 0.181 },
        radius = 10,
        callBack = showToast
    })

    mui.newTextField({
        name = "textfield_demo4",
        labelText = "My Topic",
        text = "Hello, World!",
        font = native.systemFont,
        width = 200,
        height = 30,
        x = 130,
        y = 190,
        activeColor = { 0.12, 0.67, 0.27, 1 },
        inactiveColor = { 0.4, 0.4, 0.4, 1 },
        callBack = mui.textfieldCallBack
    })

    -- horizontal slider (vertical in development)
    ---[[--
    mui.newSlider({
        name = "slider_demo",
        width = 200,
        height = 2,
        x = 130,
        y = 260,
        radius = 12,
        colorOff = { 1, 1, 1, 1 },
        color = { 0.63, 0.81, 0.181 },
        startPercent = 30,
        callBackMove = mui.sliderCallBackMove,
        callBack = mui.sliderCallBack
    })

    mui.newSlider({
        name = "slider_demo2",
        width = 200,
        height = 2,
        x = 130,
        y = 300,
        radius = 12,
        colorOff = { 1, 1, 1, 1 },
        color = { 0.31, 0.65, 0.03, 1 },
        startPercent = 60,
        callBackMove = mui.sliderCallBackMove,
        callBack = mui.sliderCallBack
    })
    --]]--

    -- Create the widget
    ---[[--
    local scrollWidth = muiData.safeAreaWidth * 0.5
    scrollView = widget.newScrollView(
        {
            top = 15,
            left = (muiData.safeAreaWidth - scrollWidth),
            width = scrollWidth,
            height = muiData.safeAreaHeight - 50,
            scrollWidth = scrollWidth,
            scrollHeight = (muiData.safeAreaHeight * 2),
            listener = scrollAListener
        }
    )

    mui.newTextField({
        name = "textfield_demo2",
        labelText = "Subject",
        text = "Hello, world!",
        font = native.systemFont,
        width = 200,
        height = 30,
        x = 120,
        y = 80,
        activeColor = { 0.12, 0.67, 0.27, 1 },
        inactiveColor = { 0.4, 0.4, 0.4, 1 },
        callBack = mui.textfieldCallBack,
        scrollView = scrollView
    })

    mui.newTextField({
        name = "textfield_demo3",
        labelText = "Tweet",
        text = "Scroll away",
        font = native.systemFont,
        width = 200,
        height = 30,
        x = 120,
        y = 160,
        activeColor = { 0.12, 0.67, 0.27, 1 },
        inactiveColor = { 0.4, 0.4, 0.4, 1 },
        callBack = mui.textfieldCallBack,
        scrollView = scrollView,
        isSecure = true
    })

    -- create a drop down list
    local numOfRowsToShow = 3
    mui.newSelect({
        name = "selector_demo2",
        labelText = "Favorite Food",
        text = "Apple",
        font = native.systemFont,
        textColor = { 0.4, 0.4, 0.4 },
        fieldBackgroundColor = { 1, 1, 1, 1 },
        rowColor = { default={ 1, 1, 1, 1 }, over={ 1, 0.5, 0, 0.2 } }, -- default is the highlighting
        rowBackgroundColor = { 1, 1, 1, 1 }, -- the drop down color of each row
        touchpointColor = { 0.4, 0.4, 0.4 }, -- the touchpoint color
        activeColor = { 0.12, 0.67, 0.27, 1 },
        inactiveColor = { 0.4, 0.4, 0.4, 1 },
        strokeColor = { 0.4, 0.4, 0.4, 1 },
        strokeWidth = 2,
        width = 200,
        height = 30,
        listHeight = 30 * numOfRowsToShow,
        x = 120,
        y = 240,
        callBackTouch = mui.onRowTouchSelector,
        scrollListener = nil,
        list = { -- if 'key' use it for 'id' in the table row
            { key = "Row1", text = "Apple", value = "Apple", isCategory = false, backgroundColor = {1,1,1,1} },
            { key = "Row2", text = "Cookie", value = "Cookie", isCategory = false },
            { key = "Row3", text = "Pizza", value = "Pizza", isCategory = false },
            { key = "Row4", text = "Shake", value = "Shake", isCategory = false },
            { key = "Row5", text = "Shake 2", value = "Shake 2", isCategory = false },
            { key = "Row6", text = "Shake 3", value = "Shake 3", isCategory = false },
            { key = "Row7", text = "Shake 4", value = "Shake 4", isCategory = false },
            { key = "Row8", text = "Shake 5", value = "Shake 5", isCategory = false },
            { key = "Row9", text = "Shake 6", value = "Shake 6", isCategory = false },
        },
        scrollView = scrollView,
    })

    --[[--
    local newlist = {}
    newlist = {
        { key = "Row1", text = "Dog", value = "Puggle", isCategory = false},
        { key = "Row2", text = "Cat", value = "Tabby", isCategory = false },
        { key = "Row3", text = "Dinosaur", value = "Raptor", isCategory = false },
    }
    mui.setSelectorList("selector_demo2", newlist)
    --]]--

    mui.newTextBox({
        name = "textbox_demo1",
        labelText = "Secret Text Box",
        text = "I am hidden in view\nYes, me too!\nFood\nDrink\nDesert\n1\n2\n3\n4\n5",
        font = native.systemFont,
        fontSize = 16,
        textBoxFontSize = 16,
        width = 200,
        height = 100,
        x = 120,
        y = 355,
        trimFakeTextAt = 80, -- trim at 1..79 characters.
        activeColor = { 0.12, 0.67, 0.27, 1 },
        inactiveColor = { 0.4, 0.4, 0.4, 1 },
        callBack = mui.textfieldCallBack,
        isEditable = true,
        doneButton = {
            width = 100,
            height = 30,
            fillColor = { 0.25, 0.75, 1, 1 },
            textColor = { 1, 1, 1 },
            text = "done",
            iconText = "done",
            iconFont = mui.materialFont,
            iconFontColor = { 1, 1, 1, 1 },
            radius = mui.getScaleX(8), -- set to 0 for newRectButton() instead of rounded
        },
        overlayBackgroundColor = { 1, 1, 1, 1 },
        overlayTextBoxBackgroundColor = { .9, .9, .9, 1 },
        overlayTextBoxHeight = 100,
        scrollView = scrollView
    })

    -- mui.setTextBoxValue("textbox_demo1", "toys in store")
    --[[--

    mui.newTextField({
        name = "textfield_demo5",
        labelText = "Numbers Only",
        text = "12345",
        font = native.systemFont,
        width = mui.getScaleVal(400),
        height = mui.getScaleVal(46),
        x = mui.getScaleVal(240),
        y = mui.getScaleVal(530),
        activeColor = { 0.12, 0.67, 0.27, 1 },
        inactiveColor = { 0.4, 0.4, 0.4, 1 },
        callBack = mui.textfieldCallBack,
        scrollView = scrollView,
        inputType = "number"
    })

    --]]--

    local textOptions =
    {
        --parent = textGroup,
        text = "Scroll the above",
        width = 150,
        font = native.systemFont,
        fontSize = 16,
        align = "left"  --new alignment parameter
    }
    infoText = display.newText( textOptions )
    infoText:setFillColor( 0.4, 0.4, 0.4 )
    infoText.x = scrollView.x + (scrollView.contentWidth * .1)
    infoText.y = muiData.safeAreaHeight - infoText.contentHeight * .5

    -- slide panel example
    local hideSlidePanel = function(event)
        mui.debug("home button pushed")
        -- or use close method below to close and release slider from memory
        -- mui.closeSlidePanel("slidepanel-demo")
        mui.hideSlidePanel("slidepanel-demo2")
    end

    -- slide panel example, uses navbar's "menu" icon below
    local showSlidePanel2 = function(event)
        mui.newSlidePanel({
            name = "slidepanel-demo2",
            title = "MUI Demo", -- leave blank for no panel title text
            titleAlign = "center",
            font = native.systemFont,
            width = 250,
            titleFontSize = 18,
            titleFontColor = { 1, 1, 1, 1 },
            titleFont = native.systemFont,
            titleBackgroundColor = { 0.63, 0.81, 0.181 },
            fontSize = 18,
            fillColor = { 1, 1, 1, 1 }, -- background color
            buttonToAnimate = "menu",
            callBack = nil,
            labelColor = { 0.63, 0.81, 0.181 }, -- active
            labelColorOff = { 0.5, 0.5, 0.5, 1 }, -- non-active
            buttonHeight = 36, -- fontSize * 2
            buttonHighlightColor = { 0.5, 0.5, 0.5 },
            buttonHighlightColorAlpha = 0.5,
             touchpoint = true,
            list = {
                { key = "Home", value = "1", icon="home", labelText="Home", isActive = true },
                { key = "Newsroom", value = "2", icon="new_releases", labelText="News", isActive = false },
                { key = "Location", value = "3", icon="location_searching", labelText="Location Information", isActive = false },
                { key = "To-do", value = "4", icon="view_list", labelText="To-do", isActive = false },
            },
        })
    end

    -- put navbar on bottom. this is to stay on top of other widgets.
    -- supported widget types are : "RRectButton", "RectButton", "IconButton", "Slider", "TextField"
    mui.newNavbar({
        name = "navbar_demo",
        --width = mui.getScaleVal(500), -- defaults to display.contentWidth
        height = 40,
        left = 0,
        top = 0,
        fillColor = { 0.63, 0.81, 0.181 },
        activeTextColor = { 1, 1, 1, 1 },
        padding = 5,
    })
    mui.newIconButton({
        name = "menu",
        text = "menu",
        width = 25,
        height = 25,
        x = 0,
        y = 0,
        font = mui.materialFont,
        textColor = { 1, 1, 1 },
        textAlign = "center",
        callBack = showSlidePanel2
    })
    mui.attachToNavBar( "navbar_demo", {
        widgetName = "menu",
        widgetType = "IconButton",
        align = "left",  -- left | right supported
    })
    mui.newIconButton({
        name = "refresh",
        text = "refresh",
        width = 25,
        height = 25,
        x = 0,
        y = 0,
        font = mui.materialFont,
        textColor = { 1, 1, 1 },
        textAlign = "center",
        callBack = mui.actionForButton
    })
    mui.attachToNavBar( "navbar_demo", {
        widgetName = "refresh",
        widgetType = "IconButton",
        align = "left",  -- left | right supported
    })
    mui.newTextField({
        name = "textfield_nav",
        text = "",
        placeholder = "Search",
        font = native.systemFont,
        width = 200,
        height = 30,
        x = 0,
        y = 0,
        activeColor = { 1, 1, 1, 1 },
        inactiveColor = { 1, 1, 1, 0.8 },
        fillColor = { 0.63, 0.81, 0.181 },
        callBack = mui.textfieldCallBack
    })
    mui.attachToNavBar( "navbar_demo", {
        widgetName = "textfield_nav",
        widgetType = "TextField",
        align = "left",  -- left | right supported
    })
    mui.newIconButton({
        name = "help",
        text = "help",
        width = 25,
        height = 25,
        x = 0,
        y = 0,
        font = mui.materialFont,
        textColor = { 1, 1, 1 },
        textAlign = "center",
        callBack = mui.actionForButton
    })
    mui.attachToNavBar( "navbar_demo", {
        widgetName = "help",
        widgetType = "IconButton",
        align = "right",  -- left | right supported
    })

    --
    -- create a Generic User-defined widget and attach to navbar
    --
    local textOptions =
    {
        --parent = textGroup,
        text = "Ready",
        x = 0,
        y = 0,
        font = native.systemFont,
        fontSize = 18,
        align = "left"  --new alignment parameter
    }
    local demoText = display.newText( textOptions )
    demoText:setFillColor( unpack( {1, 1,1 ,1} ) )
    mui.attachToNavBar( "navbar_demo", {
        widgetName = "demoText",
        widgetType = "Generic",
        widgetObject = demoText,
        destroyCallBack = destroyDemoText, -- user supplied method, must be defined otherwise it will not free memory
        padding = 12,
        align = "right",  -- left | right supported
    })

end

--
-- a generic scroll to hold ui elements
--

-- ScrollView listener
function scrollAListener( event )

    local phase = event.phase
    if event.phase == nil then return end

    mui.updateEventHandler( event )

    if ( phase == "began" ) then
        -- skip it
    elseif ( phase == "moved" ) then
        mui.updateUI(event)
    elseif ( phase == "ended" ) then
        mui.removeFocus()
        -- mui.debug( "Scroll view was released" )
    end

    -- In the event a scroll limit is reached...
    if ( event.limitReached ) then
        if ( event.direction == "up" ) then mui.debug( "Reached bottom limit" )
        elseif ( event.direction == "down" ) then mui.debug( "Reached top limit" )
        elseif ( event.direction == "left" ) then mui.debug( "Reached right limit" )
        elseif ( event.direction == "right" ) then mui.debug( "Reached left limit" )
        end
    end

    return true
end


function scene:show( event )
	local sceneGroup = self.view
	local phase = event.phase
	
	if phase == "will" then
		-- Called when the scene is still off screen and is about to move on screen
	elseif phase == "did" then
		-- Called when the scene is now on screen
		-- 
		-- INSERT code here to make the scene come alive
		-- e.g. start timers, begin animation, play audio, etc.
		--physics.start()
        -- mui.actionSwitchScene({callBackData={sceneDestination="menu"}})

	end
end

function scene:hide( event )
	local sceneGroup = self.view
	
	local phase = event.phase
	
	if event.phase == "will" then
		-- Called when the scene is on screen and is about to move off screen
		--
		-- INSERT code here to pause the scene
		-- e.g. stop timers, stop animation, unload sounds, etc.)
		--physics.stop()
	elseif phase == "did" then
		-- Called when the scene is now off screen
	end	
	
end

function scene:destroy( event )

	-- Called prior to the removal of scene's "view" (sceneGroup)
	-- 
	-- INSERT code here to cleanup the scene
	-- e.g. remove display objects, remove touch listeners, save state, etc.
	local sceneGroup = self.view

    mui.destroy()

    if scrollView ~= nil then
        scrollView:removeSelf()
        scrollView = nil
    end

    if background ~= nil then
        background:removeSelf()
        background = nil
    end

    if infoText ~= nil then
        infoText:removeSelf()
        infoText = nil
    end

	sceneGroup:removeSelf()
	sceneGroup = nil
end

---------------------------------------------------------------------------------

-- Listener setup
scene:addEventListener( "create", scene )
scene:addEventListener( "show", scene )
scene:addEventListener( "hide", scene )
scene:addEventListener( "destroy", scene )

-----------------------------------------------------------------------------------------

return scene